modules/features2d/misc/java/test/SURFDescriptorExtractorTest.java

```diff
@@ -89,20 +89,25 @@ public void testEmpty() {
 
     public void testRead() {
         String filename = OpenCVTestRunner.getTempFileName("yml");
-        writeFile(filename, "%YAML:1.0\n---\nnOctaves: 4\nnOctaveLayers: 2\nextended: 1\nupright: 0\n");
+        writeFile(filename, "%YAML:1.0\n---\nname: \"Feature2D.SURF\"\nhessianThreshold: 100.\nextended: 1\nupright: 0\nnOctaves: 4\nnOctaveLayers: 2\n");
 
         extractor.read(filename);
 
         assertEquals(128, extractor.descriptorSize());
+        //assertEquals(100., extractor.getHessianThreshold());
+        //assertEquals(1, extractor.getExtended());
+        //assertEquals(0, extractor.getUpright());
+        //assertEquals(4, extractor.getNOctaves());
+        //assertEquals(2, extractor.getNOctaveLayers());
```

Comment:
I propose to restore checks.

Reply:
This is additionnal checks I tired to add but I didn't succeed originally. Problem was the following:
 - extractor is of type "Feature2D", so It does not have access to methods of derived class SURF like getExtended()...

Here is my proposed workaround:
```patch
diff --git a/modules/features2d/misc/java/test/SURFDescriptorExtractorTest.java b/modules/features2d/misc/java/test/SURFDescriptorExtractorTest.java
index 2bbaad1e2c..ed287ea074 100644
--- a/modules/features2d/misc/java/test/SURFDescriptorExtractorTest.java
+++ b/modules/features2d/misc/java/test/SURFDescriptorExtractorTest.java
@@ -10,10 +10,11 @@ import org.opencv.test.OpenCVTestCase;
 import org.opencv.test.OpenCVTestRunner;
 import org.opencv.imgproc.Imgproc;
 import org.opencv.features2d.Feature2D;
+import org.opencv.xfeatures2d.SURF;
 
 public class SURFDescriptorExtractorTest extends OpenCVTestCase {
 
-    Feature2D extractor;
+    SURF extractor;
     int matSize;
 
     private Mat getTestImg() {
@@ -29,7 +30,7 @@ public class SURFDescriptorExtractorTest extends OpenCVTestCase {
         super.setUp();
 
         Class[] cParams = {double.class, int.class, int.class, boolean.class, boolean.class};
-        Object[] oValues = {100, 4, 2, true, false};
+        Object[] oValues = {100, 2, 4, true, false};
         extractor = createClassInstance(XFEATURES2D+"SURF", DEFAULT_FACTORY, cParams, oValues);
 
         matSize = 100;
@@ -89,16 +90,16 @@ public class SURFDescriptorExtractorTest extends OpenCVTestCase {
 
     public void testRead() {
         String filename = OpenCVTestRunner.getTempFileName("yml");
-        writeFile(filename, "%YAML:1.0\n---\nname: \"Feature2D.SURF\"\nhessianThreshold: 100.\nextended: 1\nupright: 0\nnOctaves: 4\nnOctaveLayers: 2\n");
+        writeFile(filename, "%YAML:1.0\n---\nname: \"Feature2D.SURF\"\nhessianThreshold: 100.\nextended: 1\nupright: 0\nnOctaves: 2\nnOctaveLayers: 4\n");
 
         extractor.read(filename);
 
         assertEquals(128, extractor.descriptorSize());
-        //assertEquals(100., extractor.getHessianThreshold());
-        //assertEquals(1, extractor.getExtended());
-        //assertEquals(0, extractor.getUpright());
-        //assertEquals(4, extractor.getNOctaves());
-        //assertEquals(2, extractor.getNOctaveLayers());
+        assertEquals(100., extractor.getHessianThreshold());
+        assertEquals(true, extractor.getExtended());
+        assertEquals(false, extractor.getUpright());
+        assertEquals(2, extractor.getNOctaves());
+        assertEquals(4, extractor.getNOctaveLayers());
     }
```
If you agree with this (replacing Feature2D by XXX), I can do the changes for all tests of other Features2d to really check read values with testRead() functions (until now, read values were not checked).

Reply:
It's good idea. Please go ahead.